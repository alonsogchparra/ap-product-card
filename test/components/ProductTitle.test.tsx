import React from 'react';
import { render } from '@testing-library/react';
import ProductCard, { ProductTitle } from '../../src/components';
import { product1 } from '../../src/data/products';

describe('ProductTitle', () => {
  test('should render the component correctly wit the custom title', () => {
    const { asFragment } = render(<ProductTitle title="Custom Title" />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should be shoe the component with the product name', () => {
    const { asFragment } = render(
      <ProductCard product={product1}>{() => <ProductTitle />}</ProductCard>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
