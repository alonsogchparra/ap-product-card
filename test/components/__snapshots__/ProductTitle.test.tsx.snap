// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle should be shoe the component with the product name 1`] = `
<DocumentFragment>
  <div
    class="productCard undefined"
  >
    <span
      class="productDescription undefined"
    >
      Coffee Mug - Card
    </span>
  </div>
</DocumentFragment>
`;

exports[`ProductTitle should render the component correctly wit the custom title 1`] = `
<DocumentFragment>
  <span
    class="productDescription undefined"
  >
    Custom Title
  </span>
</DocumentFragment>
`;
